### PR TITLE
Optional segyio and segy write

### PR DESCRIPTION
**Pull request by d-chambers**
## Description

This PR removes segyio as a hard dependency. Segyio is great, but usually very slow to upgrade to newer python versions which, in tern, keeps DASCore from upgrading.  This PR fixes this. 

It also implements very simple support for writing segy files as requested in #428. This will probably require more work in the future to get right, but at least we have a start here.

@aaronjgirard, do you mind reviewing this PR? 

## Checklist

I have (if applicable):

- [ ] referenced the GitHub issue this PR closes.
- [ ] documented the new feature with docstrings or appropriate doc page.
- [ ] included a test. See [testing guidelines](https://dascore.org/contributing/testing.html).
- [ ] your name has been added to the contributors page (docs/contributors.md).
- [ ] added the "ready_for_review" tag once the PR is ready to be reviewed.

### CODE REVIEW

**Comment by d-chambers** [comment]
I am particularly interested if you have ideas for improving the detection of segy files found [here](https://github.com/DASDAE/dascore/blob/2c253c6dd3aa67f9f38f2436f85d9c2737d28498/dascore/io/segy/utils.py#L39-L79).

File: dascore/io/segy/core.py [hunk 1]
```diff
@@ -55,7 +61,75 @@ def scan(self, path, **kwargs) -> list[dc.PatchAttrs]:
...
+
+        spec = segyio.spec()
+        version = self.version
+
+        # spec.sorting = 2
+        spec.format = 1  # 1 means float32 TODO look into supporting more
+        spec.samples = np.ones(len(time)) * len(channel)
+        spec.ilines = range(len(channel))
+        spec.xlines = [1]
+        # breakpoint()
```
Comment (d-chambers) [inline dascore/io/segy/core.py:94]: remove

File: dascore/io/segy/utils.py [hunk 2]
```diff
@@ -3,14 +3,80 @@
...
+
+# Valid data format codes as specified in the SEGY rev1 manual.
+VALID_FORMATS = [1, 2, 3, 4, 5, 8]
+
+# This is the maximum possible interval between two samples due to the nature
+# of the SEG Y format.
+MAX_INTERVAL_IN_SECONDS = 0.065535
+
+# largest number possible with int16
+MAX_NUMBER_OF_SAMPLES = 32767
```
Comment (d-chambers) [inline dascore/io/segy/utils.py:27]: Do we need these?

**Comment by d-chambers** [comment]
This is failing. @aaronjgirard and I discussed `spool(data_directory_with_segy).update()` should issue a warning while `spool(segy_file.segy)` should raise. In both cases the message should indicate segyio needs to be installed.

Also, for now we are just going to support converting patches with a dimension of "channel" to segy. In the future we will figure out how to handle 'distance' or 'x', 'y' coords.

**Comment by d-chambers** [comment]
We also need to add a test for standard patch conversion

File: dascore/io/segy/utils.py (hunk 2 above)
Comment (aaronjgirard) [reply dascore/io/segy/utils.py:27]: It looks like we don't since they are edge cases. But for the writer, we may want to ensure we don't make a trace with more than max number of samples. Since DAS can be very small dt and very large time, this could become an issue.

**Comment by d-chambers** [comment]
Hey @aaronjgirard,

I am going to merge this. I think I implemented everything we planned on for the first pass at segy write, and we really need this fix to support python 3.13. No doubt, we will have improvements to make to segy in the future!
